### PR TITLE
feat: support handle very large uid/gid in PAX style tar 

### PR DESCRIPTION
**Pull request by adamqqqplay**
This commit add process to read large uid/gid from PAX extensions
to fix very large UIDs/GIDs (>=2097151, limit of USTAR tar) lost
in PAX style tar during unpack.
    
Fix: #332

### CODE REVIEW

File: src/archive.rs [hunk 1]
```diff
@@ -275,7 +275,7 @@ impl<'a, R: Read> Iterator for Entries<'a, R> {
 impl<'a> EntriesFields<'a> {
     fn next_entry_raw(
         &mut self,
-        pax_size: Option<u64>,
+        pax_extensions: &Option<Vec<u8>>,
```
Comment (jiangliu) [inline src/archive.rs:278]: How about using `pax_extensions: Option<&[u8]>>`?

File: src/archive.rs [hunk 2]
```diff
@@ -315,13 +315,31 @@ impl<'a> EntriesFields<'a> {
             return Err(other("archive header checksum mismatch"));
         }
 
+        let mut pax_size: Option<u64> = None;
+        let mut pax_uid: Option<u64> = None;
+        let mut pax_gid: Option<u64> = None;
+        if let Some(pax_extensions_ref) = &pax_extensions {
+            pax_size = pax_extensions_value(pax_extensions_ref, PAX_SIZE);
+            pax_uid = pax_extensions_value(pax_extensions_ref, PAX_UID);
```
Comment (jiangliu) [inline src/archive.rs:323]: How about
```
if let Some(uid) = pax_extensions_value(pax_extensions_ref, PAX_UID) {
       header.set_uid(uid);
}
```

File: src/archive.rs (hunk 1 above)
Comment (adamqqqplay) [reply src/archive.rs]: Fixed, thanks.

File: src/archive.rs (hunk 2 above)
Comment (adamqqqplay) [reply src/archive.rs]: Fixed.

**Comment by adamqqqplay** [comment]
@alexcrichton Hi Alex, could you take a look when you have time? Thanks!

**Review by alexcrichton — APPROVED** [review]
Thanks!

**Comment by adamqqqplay** [comment]
@alexcrichton Thanks for your review!

**Pull request merged**